app/services/assertions.rb

```diff
@@ -0,0 +1,42 @@
+module Assertions
+  class Error < StandardError
+    def initialize(assertion)
+      super("Assertion failed: #{assertion}")
+    end
+  end
+
+  class ErrorWithEdition < Error
```

Comment:
This feels these errors should be more specific i.e. Edition::NotEditableError,  Edition::BackdateRemovalError etc... (probably can think of better names) 

This would provide better descriptions of what happened and allow for more specific control flow depending on the issue.